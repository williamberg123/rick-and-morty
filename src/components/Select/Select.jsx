import React from 'react';
import PropTypes from 'prop-types';

import './Select.css';

export default function Select({ funcChange }){
    return (
        <select className="Select" onChange={funcChange}>
            <option value="personagens">PERSONAGENS</option>
            <option value="episodios">EPISÃ“DIOS</option>
            <option value="ambientes">AMBIENTES</option>
        </select>
    );
}

Select.propTypes = {
    funcChange: PropTypes.func.isRequired
};