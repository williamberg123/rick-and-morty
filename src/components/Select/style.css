.Select {
    justify-self: start;
    width: 300px;
    height: 40px;
    margin: 10px auto 20px 10px;
}

@media (max-width: 700px) {
    .Select {
        margin: 10px 0;
    }
}

@media (max-width: 450px) {
    .Select {
        width: 100%;
    }
}