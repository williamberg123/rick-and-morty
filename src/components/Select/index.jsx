import React, { memo, useContext } from 'react';

import AppContext from '../../AppContext';

import './style.css';

function Select(){
    const { handleChangeOption } = useContext(AppContext);

    return (
        <select className="Select" onChange={handleChangeOption}>
            <option value="personagens">PERSONAGENS</option>
            <option value="episodios">EPISÃ“DIOS</option>
            <option value="ambientes">AMBIENTES</option>
        </select>
    );
}

export default memo(Select);