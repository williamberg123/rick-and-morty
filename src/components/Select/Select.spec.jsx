import React from 'react';
import { screen, render, fireEvent } from '@testing-library/react';
import Select from './index';

describe('<Select />', () => {
    it('should render select with options', () => {
        const fn = jest.fn();
        render(<Select funcChange={fn} />);

        expect(screen.getAllByRole('option')).toHaveLength(3);
    });

    it('should call function when a option is change', () => {
        const fn = jest.fn();
        render(<Select funcChange={fn} />);

        fireEvent.change(screen.getByRole('option', { name: /personagens/i }).parentElement);

        expect(fn).toBeCalledTimes(1);
    });

    it('should match snapshot', () => {
        const fn = jest.fn();
        render(<Select funcChange={fn} />);
        
        expect(screen.getByRole('option', { name: /personagens/i }).parentElement)
            .toMatchSnapshot();
    });
});

// Testado