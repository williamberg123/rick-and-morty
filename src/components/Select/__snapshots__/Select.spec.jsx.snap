// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select /> should match snapshot 1`] = `
<select
  class="Select"
>
  <option
    value="personagens"
  >
    PERSONAGENS
  </option>
  <option
    value="episodios"
  >
    EPISÃ“DIOS
  </option>
  <option
    value="ambientes"
  >
    AMBIENTES
  </option>
</select>
`;
